libraries/core_libs/consensus/src/pbft/block_proposer.cpp

```diff
@@ -104,10 +131,11 @@ void BlockProposer::start() {
       if (auto net = network_.lock()) {
         syncing = net->pbft_syncing();
       }
-      if (!syncing) {
-        propose_model_->propose();
+      // Only sleep if block was not proposed, if block is proposed try to propose another block immediately
```

Comment:
I think the code in line 135 is correct, but the comment is wrong.
I think should be if propose successful need take a sleep, because may take process for new incoming trxs. If not proposed, that will try another chance immediately.

Reply:
I think we should never delay proposal if we might be able to propose. Only if we cannot propose than we sleep. There is a decent chance that there are new transactions since we proposed the first block